inference/core/workflows/core_steps/sinks/roboflow/model_monitoring_exporter/v1.py

```diff
@@ -0,0 +1,271 @@
+import hashlib
+import logging
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime
+from functools import partial
+from typing import List, Literal, Optional, Tuple, Type, Union
+
+import supervision as sv
+from fastapi import BackgroundTasks
+from pydantic import ConfigDict, Field
+
+from inference.core.cache.base import BaseCache
+from inference.core.env import DEVICE_ID
+from inference.core.managers.metrics import get_system_info
+from inference.core.roboflow_api import (
+    export_inference_to_model_monitoring,
+    get_roboflow_workspace,
+)
+from inference.core.workflows.execution_engine.constants import INFERENCE_ID_KEY
+from inference.core.workflows.execution_engine.entities.base import OutputDefinition
+from inference.core.workflows.execution_engine.entities.types import (
+    BOOLEAN_KIND,
+    CLASSIFICATION_PREDICTION_KIND,
+    INSTANCE_SEGMENTATION_PREDICTION_KIND,
+    KEYPOINT_DETECTION_PREDICTION_KIND,
+    OBJECT_DETECTION_PREDICTION_KIND,
+    STRING_KIND,
+    StepOutputSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    BlockResult,
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+
+LAST_REPORT_TIME_CACHE_KEY = "roboflow_model_monitoring_last_report_time"
+WORKSPACE_NAME_CACHE_EXPIRE = 900  # 15 min
+
+SHORT_DESCRIPTION = "Periodically export inference results to Roboflow Model Monitoring"
+
+LONG_DESCRIPTION = """
+Block allows users to report inference results to Roboflow Model Monitoring.
+
+This is useful for reporting results to Model Monitoring when using InferencePipeline, which
+does not automatically report results to Model Monitoring. This block only reports back a sample
+of the results, which are specified by the `frequency` parameter.
+
+For more information on Model Monitoring at Roboflow, see https://docs.roboflow.com/deploy/model-monitoring.
+"""
+
+BLOCK_NAME = "Roboflow Model Monitoring Exporter"
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": BLOCK_NAME,
+            "version": "v1",
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "sink",
+        }
+    )
+    type: Literal[
+        "roboflow_core/roboflow_model_monitoring_exporter@v1",
+        BLOCK_NAME,
+    ]
+    predictions: StepOutputSelector(
+        kind=[
+            OBJECT_DETECTION_PREDICTION_KIND,
+            INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            KEYPOINT_DETECTION_PREDICTION_KIND,
+            CLASSIFICATION_PREDICTION_KIND,
+        ]
+    ) = Field(
+        description="Reference data to extract property from",
+        examples=["$steps.my_step.predictions"],
+    )
+    frequency: Union[
```

Comment:
I considered using the RateLimiter block, but this export block would always require that as a dependency. I'm not sure if there's a pattern for making another block required to run, so I just added this to the params for now, open to doing it another way

Reply:
Yeah maybe something like SamplePredictions, ReportPredictionsSample, ModelMonitoringPredictionsSampler